src/main/java/org/reactome/release/orthopairs/UniProtGeneNamesRetriever.java

```diff
@@ -129,25 +134,58 @@ public static void retrieveAndStoreGeneNameMappings(String speciesKey, String re
         int count = 0;
         Set<String> uniprotAccessionsToGeneNames = new HashSet<>();
         for (Set<String> uniprotIdentifierPartition : partitionedUniProtIds) {
+            int currentAttemptNum = 0;
             // Build UniProt API query from Set of 250 UniProt identifiers.
             Query query = UniProtQueryBuilder.accessions(uniprotIdentifierPartition);
             // Perform UniProt API query to retrieve gene names associated with identifiers.
-            QueryResult<UniProtComponent<Gene>> uniprotEntries = uniprotService.getGenes(query);
-
-            while (uniprotEntries.hasNext()) {
-                count++;
-                // Get Gene object returned from UniProt.
-                UniProtComponent<Gene> geneObject = uniprotEntries.next();
-                if (!geneObject.getComponent().isEmpty()) {
-                    // Iterate through all Gene components in the response.
-                    for (Gene geneComponent : geneObject.getComponent()) {
-                        // Tab-separate UniProt accession ID and its associated gene name, and then store these in the Set that will be returned.
-                        uniprotAccessionsToGeneNames.add(geneObject.getAccession().toString() + "\t" + geneComponent.getGeneName().toString() + "\n");
+            QueryResult<UniProtComponent<Gene>> uniprotEntries = null;
+            while (currentAttemptNum < MAX_NUM_ATTEMPTS && uniprotEntries == null)
```

Comment:
Perhaps after this loop if `uniprotEntries` is still null (because the maximum number of attempts is gone past), a logger error to indicate that the maximum number of attempts were surpassed and no entries were retrieved (in addition to the log for an error during retrieval).